ambry-store/src/main/java/com.github.ambry.store/BlobStoreCompactor.java

```diff
@@ -643,8 +643,8 @@ private boolean copyDataByIndexSegment(LogSegment logSegmentToCopy, IndexSegment
           logger.trace("Fetching the PUT entry of a deleted blob with entry {} in index segment with start offset {} in"
               + " {} because it needs to be retained", indexEntry, indexSegmentStartOffset, storeId);
           long putRecordOffset = value.getOriginalMessageOffset();
-          if (putRecordOffset != IndexValue.UNKNOWN_ORIGINAL_MESSAGE_OFFSET
-              && indexSegmentStartOffset.getOffset() <= putRecordOffset) {
+          if (putRecordOffset != IndexValue.UNKNOWN_ORIGINAL_MESSAGE_OFFSET && putRecordOffset != value.getOffset()
```

Comment:
Cleared original message offset for orphaned delete records as suggested - however, this code needs to be around because the old recovery code (before the bug) was not doing this.